redash/authentication/ldap_auth.py

```diff
@@ -1,26 +1,29 @@
 import logging
-logger = logging.getLogger('ldap_auth')
 
 from redash import settings
 
 from flask import flash, redirect, render_template, request, url_for, Blueprint
-from flask_login import current_user, login_required, login_user, logout_user
+from flask_login import current_user
 
 try:
-    from ldap3 import Server, Connection, SIMPLE, ANONYMOUS, NTLM
+    from ldap3 import Server, Connection
 except ImportError:
     if settings.LDAP_LOGIN_ENABLED:
-        logger.error("The ldap3 library was not found. This is required to use LDAP authentication (see requirements.txt).")
-        exit()
+        raise RuntimeError(
+            "The ldap3 library was not found. This is required to use LDAP authentication (see requirements.txt)."
+        )
```

Comment:
Just a tiny correction to follow Python best practices, instead of using the logging system for something that Python is able to handle more graceful via the RuntimeError.

Reply:
Fair enough, let's use `sys.exit` straight away since that solves the immediate problem of the logger being configurable (and possibly hiding the message). For the record, the `exit` function isn't a good practice since it's [only supposed to be used for interactive sessions](https://docs.python.org/3/library/constants.html#exit), not general purpose Python scripts. `sys.exit` on the other hand will [return an error code of 1 with the message printed to stderr](https://docs.python.org/3/library/sys.html#sys.exit).